core/src/main/java/hudson/tasks/ArtifactArchiver.java

```diff
@@ -141,9 +141,9 @@ public ArtifactArchiver(String artifacts, String excludes, boolean latestOnly, b
     }
 
     // Backwards compatibility for older builds
-    @SuppressFBWarnings(value = "RCN_REDUNDANT_NULLCHECK_OF_NONNULL_VALUE", 
+    @SuppressFBWarnings(value = "RCN_REDUNDANT_NULLCHECK_OF_NONNULL_VALUE",
             justification = "Null checks in readResolve are valid since we deserialize and upgrade objects")
-    public Object readResolve() {
+    protected Object readResolve() {
```

Comment:
We should just make classes like this `final` anyway. They were never designed for subclassing and any plugin attempting to do so is likely to get broken in all sorts of ways anyway.